File: week11/week11-2.py
# week11-2.py 篩子法
BOUND=20000 # 30萬以下有幾個質數?
table=[True]*BOUND
ans=0
for i in range(2,BOUND):
  if(table[i])==True:
    # print(i,end=' ')
    ans+=1 # 找到1個質數
    for k in range(i*i,BOUND,i):
      table[k]=False
print('質數有:',ans,'個')

File: week11/week11-2.cpp
#include <stdio.h>
int main()
{
    int BOUND=20000,ans=0;
    int table[20000]={}; /// 給初始，都補0
    for(int i=2;i<BOUND;i++){
        if(table[i]==0){ ///還留著，沒有被殺掉
            ans++;
            for(int k=i*i;k<BOUND;k+=i){
                table[k]=-1;
            }
        }
    }
    printf("質數有:%d 個\n",ans);
}
